### PR TITLE
ON UPDATE and ON DELETE constraint referential actions

### PR DESCRIPTION
**Pull request by Anakros**
https://github.com/vapor/vapor/issues/1312

### CODE REVIEW

File: Sources/Fluent/SQL/GeneralSQLSerializer.swift [hunk 1]
```diff
@@ -377,7 +377,7 @@ open class GeneralSQLSerializer<E: Entity>: SQLSerializer {
         case .raw(let string, _):
             return string
         case .some(let foreignKey):
-            return "CONSTRAINT \(escape(foreignKey.name)) FOREIGN KEY (\(escape(foreignKey.field))) REFERENCES \(escape(foreignKey.foreignEntity.entity)) (\(escape(foreignKey.foreignField)))"
+            return "CONSTRAINT \(escape(foreignKey.name)) FOREIGN KEY (\(escape(foreignKey.field))) REFERENCES \(escape(foreignKey.foreignEntity.entity)) (\(escape(foreignKey.foreignField))) ON UPDATE \(foreignKey.onUpdate.rawValue) ON DELETE \(foreignKey.onDelete.rawValue)"
```
Comment (tanner0101) [inline Sources/Fluent/SQL/GeneralSQLSerializer.swift]: We should not append the `ON UPDATE` or `ON DELETE` at all if they are set to `.noAction`.

**Comment by vzsg** [comment]
Both [SQLite](https://www.sqlite.org/syntax/foreign-key-clause.html) and [PostgeSQL](https://www.postgresql.org/docs/9.1/static/sql-createtable.html) support the ON UPDATE/ON DELETE syntax.

**Comment by tanner0101** [comment]
Great! Thanks @vzsg. Then this LGTM once the review comment is sorted.

**Comment by tanner0101** [comment]
There is still one outstanding change that needs to be sorted.

I have implemented this feature for Fluent 3. https://github.com/vapor/fluent/pull/376 

Given we're about a week away from the Vapor 3 GM release maybe we should re-consider if we need to include this in Vapor 2 as well.

**Pull request closed without merging**